docs/rfcs/env-vars.md

```diff
@@ -0,0 +1,216 @@
+
+These both have **the same character set and behavior**. They both use
+the env provider under the hood.
+
+The naked syntax supported in BASH is not supported in the Collector.
+Escaping is supported by using two dollar signs. Escaping is also
+honored for unsupported identifiers like `${1}`.
+
+### Type casting rules
+
+The environment variable value is parsed by the yaml.v3 parser to an
+any-typed variable and the original representation as a string is stored
+for numeric types. The `yaml.v3` parser mostly follows the YAML v1.2
+specification with [*some
+exceptions*](https://github.com/go-yaml/yaml#compatibility). You can see
+how it works for some edge cases in this example:
+[*https://go.dev/play/p/RtPmH8aZA1X*](https://go.dev/play/p/RtPmH8aZA1X).
+
+When unmarshalling, we use mapstructure with WeaklyTypedInput
+**disabled**. We check via a hook an `AsString` method from confmap.Conf
+and use its return value when it is valid and we are mapping to a string
+field. This method has default casting rules for unambiguous scalar
+types but may return the original representation depending on the
+construction of confmap.Conf
+
+For using this notation in inline mode (e.g.`http://endpoint/${env:PATH}`), we
+use the `AsString` method from confmap.Conf.
+
+### Character set
+
+An environment variable identifier must be a nonempty ASCII alphanumeric
+or underscore starting with an alphabetic or underscore character. Its
+maximum length is 200 characters. Both syntaxes support recursive
+resolution.
+
+When an invalid identifier is found, a warning log is emitted, and the
+string is kept as-is.
+
+### Comparison table with current behavior
+
+This is a comparison between the current and desired behavior for
+loading a field with the braces syntax, `env` syntax.
+
+| Raw value    | Field type | Current behavior, `${ENV}`, single field | Current behavior,  `${env:ENV}` , single field | Desired behavior, entire field | Desired behavior, inline string field |
+|--------------|------------|------------------------------------------|------------------------------------------------|--------------------------------|---------------------------------------|
+| `123`        | integer    | 123                                      | 123                                            | 123                            | n/a                                   |
+| `0123`       | integer    | 83                                       | 83                                             | 83                             | n/a                                   |
+| `0123`       | string     | 0123                                     | 83                                             | 0123                           | 0123                                  |
+| `0xdeadbeef` | string     | 0xdeadbeef                               | 3735928559                                     | 0xdeadbeef                     | 0xdeadbeef                            |
```

Comment:
We had some discussion about the desired behavior when setting the entire field at KubeCon EU (cc @evan-bradley). Should this be an error instead? Should we use base 10 notation?

Reply:
I've been mulling over this some more. I think the desired behavior expressed here is the right way to go.